packages/calcite-components/src/components/list/list.tsx

```diff
@@ -937,20 +954,44 @@ export class List extends LitElement implements InteractiveComponent, SortableCo
     const toEl = moveTo.element as List["el"];
     const fromElItems = Array.from(fromEl.children).filter(isListItem);
     const oldIndex = fromElItems.indexOf(dragEl);
+    const newIndex = 0;
```

Comment:
If a put/pull fails, does the index get modified to 0?

Reply:
When moving via keyboard, we always move to index 0. So that is what the fail event would show.

Reply:
Ah, gotcha. So this is keyboard-specific moving logic.

Reply:
> So this is keyboard-specific moving logic.

Scratch that. This is move-event-specific logic.

Reply:
Both, its logic specific to moving via keyboard. When moving via keyboard we always place at the top.

Reply:
Is a move is handled the same way whether it's keyboard or mouse-initiated? I guess I'm trying to confirm whether `newIndex` is updated differently when it's not moved via keyboard.

Reply:
> Is a move is handled the same way whether it's keyboard or mouse-initiated?

Not exactly, when using a mouse the sortable library lets you move to any index within another list whereas with the keyboard moving via the handle component, we only allow moving to the top of another list. Then a user would need to sort within that list to sort within the list. 

This logic is only for moving via the sort handle which places the element at the top of the list being moved to.

Reply:
Sweet. Thanks for clarifying.